Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -89,24 +89,47 @@ extension HTTPClient {
     /// Represent HTTP request.
     public struct Request {
         /// Request HTTP method, defaults to `GET`.
-        public let method: HTTPMethod
+        public var method: HTTPMethod {
+            return value.method
+        }
         /// Remote URL.
-        public let url: URL
+        public var url: URL {
+            return value.url
+        }
         /// Remote HTTP scheme, resolved from `URL`.
-        public let scheme: String
+        public var scheme: String {
+            return value.scheme
+        }
         /// Remote host, resolved from `URL`.
-        public let host: String
+        public var host: String {
+            return value.host
+        }
         /// Request custom HTTP Headers, defaults to no headers.
-        public var headers: HTTPHeaders
+        public var headers: HTTPHeaders {
+            get {
+                return value.headers
+            }
+            set {
+                value.headers = newValue
+            }
+        }
         /// Request body, defaults to no body.
-        public var body: Body?
+        public var body: Body? {
+            get {
+                return value.body
+            }
+            set {
+                value.body = newValue
+            }
+        }
 
         struct RedirectState {
             var count: Int
             var visited: Set<URL>?
         }
 
         var redirectState: RedirectState?
+        private var value: HTTPRequest
```

Comment:
I don't love using a protocol existential here. I think the splitting of basic request types is a good idea, but I'd rather use an enumeration that we can switch over than the protocol existential.

Reply:
It seems the only difference between implementations is how they handle scheme? Why not just add a `unix://` to a list of supported schemes?

Reply:
> Why not just add a unix:// to a list of supported schemes?

slightly different asserts in Request constructor

Reply:
>  I'd rather use an enumeration that we can switch over than the protocol existential.

I'm not a fan. Do you think about if'ing in every `Request` property?